packages/fxa-settings/src/lib/url.test.ts

```diff
@@ -0,0 +1,231 @@
+  });
+
+  it('returns empty if parameter is empty', () => {
+    expect(searchParam('color', '?color=')).toStrictEqual('');
+  });
+
+  it('returns empty if parameter is a space', () => {
+    expect(searchParam('color', '?color= ')).toStrictEqual('');
+  });
+
+  it('returns undefined if parameter does not exist', () => {
+    expect(searchParam('animal', '?color=green')).toBeUndefined();
+  });
+});
+
+describe('searchParams', () => {
+  const search = `?color=green&email=${encodeURIComponent(
+    'testuser@testuser.com'
+  )}#color=brown&email=${encodeURIComponent('hash@testuser.com')}`;
+
+  it('converts search string to an object, returns all key/value pairs if no allowlist specified', () => {
+    const params = searchParams(search);
+    expect(params.color).toStrictEqual('green');
+    expect(params.email).toStrictEqual('testuser@testuser.com');
+  });
+
+  it('returns only items in allow list of one is specified', () => {
+    const params = searchParams(search, ['color', 'notDefined']);
+    expect(params.color).toStrictEqual('green');
+    expect('email' in params).toBeFalsy();
+    expect('notDefined' in params).toBeFalsy();
+  });
+
+  it('returns an empty object if no query params', () => {
+    let params = searchParams('');
+    expect(Object.keys(params)).toHaveLength(0);
+
+    params = searchParams('', ['blue']);
+    expect(Object.keys(params)).toHaveLength(0);
+  });
+});
+
+describe('hashParams', () => {
+  const search = `?color=green&email=${encodeURIComponent(
+    'testuser@testuser.com'
+  )}#color=brown&email=${encodeURIComponent('hash@testuser.com')}`;
+
+  it('converts hash string to an object, returns all key/value pairs if no allowlist specified', () => {
+    const params = hashParams(search);
+    expect(params.color).toStrictEqual('brown');
+    expect(params.email).toStrictEqual('hash@testuser.com');
+  });
+
+  it('returns only items in allow list of one is specified', () => {
+    const params = hashParams(search, ['color', 'notDefined']);
+    expect(params.color).toStrictEqual('brown');
+    expect('email' in params).toBeFalsy();
+    expect('notDefined' in params).toBeFalsy();
+  });
+
+  it('returns an empty object if no hash params', () => {
+    let params = hashParams('');
+    expect(Object.keys(params)).toHaveLength(0);
+
+    params = hashParams('', ['blue']);
+    expect(Object.keys(params)).toHaveLength(0);
+  });
+});
+
+describe('objToSearchString', () => {
+  it('includes all keys with values', () => {
+    var params = {
+      hasValue: 'value',
+      nullNotIncluded: null,
+      undefinedNotIncluded: undefined,
+    };
+
+    expect(objToSearchString(params)).toStrictEqual('?hasValue=value');
+  });
+
+  it('returns an empty string if no parameters are passed in', () => {
+    expect(objToSearchString({})).toStrictEqual('');
+  });
+});
+
+describe('objToHashString', () => {
+  it('includes all keys with values', () => {
+    var params = {
```

Comment:
`var`? (there's several in this file)

Reply:
Sorry, these are ported over from the content server. Will update.